Content.Shared/Singularity/Events/SingularityLevelChangedEvent.cs

```diff
@@ -0,0 +1,38 @@
+using Content.Shared.Singularity.Components;
+using Content.Shared.Singularity.EntitySystems;
+
+namespace Content.Shared.Singularity.Events;
+
+/// <summary>
+/// An event raised whenever a singularity changes its level.
+/// </summary>
+public sealed class SingularityLevelChangedEvent : EntityEventArgs
+{
+    /// <summary>
+    /// The new level of the singularity.
+    /// </summary>
+    public readonly ulong NewValue;
+
+    /// <summary>
+    /// The previous level of the singularity.
+    /// </summary>
+    public readonly ulong OldValue;
+
+    /// <summary>
+    /// The singularity that just changed level.
+    /// </summary>
+    public readonly SharedSingularityComponent Singularity;
+
+    /// <summary>
+    /// The system that managed the level change.
+    /// </summary>
+    public readonly SharedSingularitySystem SingularitySystem;
```

Comment:
why is the system being passed in the event? people subscribing to the event can just have a dependency on the system anyway?

Reply:
Done